### PR TITLE
reduce disk usage of thirdparty build directories

### PR DESCRIPTION
**Pull request by benoit-pierre**
- shallower clones
- sparse checkouts for the initial repos
- support not fetching/setting up any submodules
- disable all OpenSSL submodules (not used)
- only fetch/checkout the MuPDF submodules that are needed

Stats after building 3 versions (emulator debug & release versions, kindlepw2 release version):

|                                                  | master |     PR |
| -------------------------                        | ------:|-------:|
| `thirdparty/*/build/git_checkout`                |   2.6G |   296M |
| `thirdparty/*/build/x86_64-pc-linux-gnu`         |   3.2G |   1.5G |
| `thirdparty/*/build/x86_64-pc-linux-gnu-debug`   |   3.6G |   2.0G |
| `thirdparty/*/build/arm-kindlepw2-linux-gnueabi` |   3.5G |   1.8G |
| `thirdparty/*/build`                             |  13.0G |   5.5G |

Still high, but much better.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1578)

### CODE REVIEW

File: thirdparty/cmake_modules/koreader_thirdparty_git.cmake [hunk 1]
```diff
@@ -69,79 +60,76 @@ if(should_clone)
   if(error_code)
     message(FATAL_ERROR \"Failed to clone repository: '${git_repository}'\")
   endif()
+  execute_process(
+    COMMAND \"${git_EXECUTABLE}\" -C \"${clone_checkout}\" sparse-checkout init
```
Comment (Frenzie) [inline thirdparty/cmake_modules/koreader_thirdparty_git.cmake:64]: This isn't supported yet. But thanks for providing the update.

**Review by Frenzie — CHANGES_REQUESTED** [review]
Not about this PR which is exactly according to plan but blocked until the Android image is updated.

File: thirdparty/cmake_modules/koreader_thirdparty_git.cmake (hunk 1 above)
Comment (benoit-pierre) [reply thirdparty/cmake_modules/koreader_thirdparty_git.cmake:64]: Note: the code should be robust to `sparse-checkout` not being supported (exit codes are ignored for that reason).

**Comment by Frenzie** [comment]
It's a bit unclear to me how you managed to make CircleCI not run. Do you have any idea?

**Comment by Frenzie** [comment]
Eh well, I think it's fine with a quick try, though it's lacking the peace of mind of having a multitude of configurations checked.

Now running on master: https://github.com/koreader/koreader-base/runs/11450672853

**Comment by benoit-pierre** [comment]
> It's a bit unclear to me how you managed to make CircleCI not run. Do you have any idea?

No idea, but from past experience Circle CI can be a bit funky.

**Comment by benoit-pierre** [comment]
Have you considered converting those jobs to use Github Actions?

**Comment by benoit-pierre** [comment]
No looking good on master...

**Comment by benoit-pierre** [comment]
I'm looking into it, seems like mupdf is missing some variables (e.g. `SYS_LIBJPEG_LIBS`).

**Comment by Frenzie** [comment]
> Have you considered converting those jobs to use Github Actions?

GitHub Actions is unfortunately not fit for purpose. See https://github.com/koreader/koreader-base/pull/1442#issuecomment-1004197219 and https://github.com/koreader/koreader-base/pull/1464.

**Pull request merged**